plugins/manarefresher/delegation_receiver.go

```diff
@@ -0,0 +1,100 @@
+package manarefresher
+
+import (
+	"sync"
+	"time"
+
+	"github.com/iotaledger/goshimmer/packages/clock"
+	"github.com/iotaledger/goshimmer/packages/ledgerstate"
+	"github.com/iotaledger/goshimmer/plugins/messagelayer"
+)
+
+// DelegationReceiver checks for delegation outputs on the wallet address and keeps the most recent delegated balance.
+type DelegationReceiver struct {
+	*wallet
+	delegatedFunds map[ledgerstate.Color]uint64
+	delFundsMutex  sync.RWMutex
+
+	sync.RWMutex
+}
+
+// Scan scans for unspent delegation outputs on the delegation receiver address
+func (d *DelegationReceiver) Scan() []*ledgerstate.AliasOutput {
+	d.Lock()
+	defer d.Unlock()
+	cachedOutputs := messagelayer.Tangle().LedgerState.OutputsOnAddress(d.address)
+	defer cachedOutputs.Release()
+	now := clock.SyncedTime()
+
+	filtered := cachedOutputs.Unwrap().Filter(func(output ledgerstate.Output) bool {
+		// it has to be an alias
+		if output.Type() != ledgerstate.AliasOutputType {
+			return false
+		}
+		// it has to be unspent
+		isUnspent := false
+		isConfirmed := false
+		messagelayer.Tangle().LedgerState.OutputMetadata(output.ID()).Consume(func(outputMetadata *ledgerstate.OutputMetadata) {
+			isUnspent = outputMetadata.ConsumerCount() == 0
+			incState, err := messagelayer.Tangle().LedgerState.TransactionInclusionState(output.ID().TransactionID())
+			if err != nil {
+				return
```

Comment:
If `messagelayer.Tangle().LedgerState.TransactionInclusionState()` returns an error it is because either its transaction is missing, or its branch. Both mean that the output is not confirmed, so it is okay to drop it. Note, that this `return` will only return from the `Consume()` function, not even from `Filter()`.